File: Wrappers.js
class LeituraNumero {
    constructor() {
        this.numeroPrimitivo = null;
        this.numeroWrapper = null;
    }

    lerNumero() {
        // Leitura do valor numérico do console
        this.numeroPrimitivo = Number(prompt("Digite um valor numérico: "));
    }

    converterParaWrapper() {
        // Conversão para o tipo Wrapper
        this.numeroWrapper = this.numeroPrimitivo;
    }

    imprimirNumero() {
        // Impressão do valor numérico
        console.log("O valor numérico é: " + this.numeroWrapper);
    }
}

const leituraNumero = new LeituraNumero();

leituraNumero.lerNumero();
leituraNumero.converterParaWrapper();
leituraNumero.imprimirNumero();

File: Wrappers.java
public class LeituraNumero {

    private double numeroPrimitivo;
    private Double numeroWrapper;
  
    public LeituraNumero() {
      this.numeroPrimitivo = 0;
      this.numeroWrapper = null;
    }
  
    public void lerNumero() {
      // Leitura do valor numérico do console
      this.numeroPrimitivo = Double.parseDouble(JOptionPane.showInputDialog("Digite um valor numérico: "));
    }
  
    public void converterParaWrapper() {
      // Conversão para o tipo Wrapper
      this.numeroWrapper = this.numeroPrimitivo;
    }
  
    public void imprimirNumero() {
      // Impressão do valor numérico
      System.out.println("O valor numérico é: " + this.numeroWrapper);
    }
  
    public static void main(String[] args) {
      LeituraNumero leituraNumero = new LeituraNumero();
  
      leituraNumero.lerNumero();
      leituraNumero.converterParaWrapper();
      leituraNumero.imprimirNumero();
    }
  }
  